integration-cli/docker_cli_events_test.go

```diff
@@ -563,3 +572,218 @@ func (s *DockerSuite) TestEventsStreaming(c *check.C) {
+	c.Assert(waitRun(cID), check.IsNil)
+
+	cmd := exec.Command(dockerBinary, "commit", "-m", "test", cID)
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't commit: %s\n%q", out, err)
+	}
+
+	cmd = exec.Command(dockerBinary, "stop", cID)
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't stop: %s\n%q", out, err)
+	}
+
+	cmd = exec.Command(dockerBinary, "events", "-f", "container="+cID, "--until="+strconv.Itoa(int(since)))
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't get events: %s\n%q", out, err)
+	}
+
+	if !strings.Contains(out, " commit\n") {
+		c.Fatalf("Missing 'commit' log event\n%s", out)
+	}
+}
+
+func (s *DockerSuite) TestEventsCopy(c *check.C) {
+	since := daemonTime(c).Unix()
+
+	id, err := buildImage("cpimg", `
+		  FROM busybox
+		  RUN echo HI > /tmp/file`, true)
+	if err != nil {
+		c.Fatalf("Couldn't create image: %q", err)
+	}
+
+	runCmd := exec.Command(dockerBinary, "run", "--name=cptest", id, "true")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		c.Fatalf("Couldn't run top: %s\n%q", out, err)
+	}
+
+	cmd := exec.Command(dockerBinary, "cp", "cptest:/tmp/file", "-")
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Failed getting file:%q\n%q", out, err)
+	}
+
+	cmd = exec.Command(dockerBinary, "events", "-f", "container=cptest", "--until="+strconv.Itoa(int(since)))
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't get events: %s\n%q", out, err)
+	}
+
+	if !strings.Contains(out, " copy\n") {
+		c.Fatalf("Missing 'copy' log event\n%s", out)
+	}
+}
+
+func (s *DockerSuite) TestEventsResize(c *check.C) {
+	since := daemonTime(c).Unix()
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "top")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		c.Fatalf("Couldn't run top: %s\n%q", out, err)
+	}
+	cID := strings.TrimSpace(out)
+	c.Assert(waitRun(cID), check.IsNil)
+
+	endpoint := "/containers/" + cID + "/resize?h=80&w=24"
+	status, _, err := sockRequest("POST", endpoint, nil)
+	c.Assert(status, check.Equals, http.StatusOK)
+	c.Assert(err, check.IsNil)
+
+	cmd := exec.Command(dockerBinary, "stop", cID)
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't stop: %s\n%q", out, err)
+	}
+
+	cmd = exec.Command(dockerBinary, "events", "-f", "container="+cID, "--until="+strconv.Itoa(int(since)))
+	out, _, err = runCommandWithOutput(cmd)
+	if err != nil {
+		c.Fatalf("Couldn't get events: %s\n%q", out, err)
+	}
+
+	if !strings.Contains(out, " resize\n") {
+		c.Fatalf("Missing 'resize' log event\n%s", out)
+	}
+}
+
+func (s *DockerSuite) TestEventsAttach(c *check.C) {
+	since := daemonTime(c).Unix()
+
+	out, _ := dockerCmd(c, "run", "-di", "busybox", "/bin/cat")
+	cID := strings.TrimSpace(out)
+
+	cmd := exec.Command(dockerBinary, "attach", cID)
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
```

Comment:
Yea, I would normally print a nicer error msg along with it but I copy-n-pasted this code from someplace else.  I'll switch it to just an assert since we don't show a nicer msg